### DISCUSSION TITLE
Unable to use Style in xaml

### DISCUSSION THREAD

**Discussion by ObsidianSilence** [canton7/Stylet#232] [Q&A]
This is my first project using Stylet so I am probably missing something. I am opening a number of views that open other views, all of the views have a button that opens another view and a GoBack button.  When I have all of the styling information in each button the project compiles and runs just fine.   
When I pull the styling values out into Styles.xaml and use the Style tag 
![image](https://user-images.githubusercontent.com/50368420/111193658-c3b2ec80-8590-11eb-845c-fa417cd4f003.png)
I get the following error message
"{DependencyProperty.UnsetValue}" is not a valid value for property 'Foreground'.
![image](https://user-images.githubusercontent.com/50368420/111193543-a4b45a80-8590-11eb-960b-ebfe5464583e.png)

**Comment by canton7** [comment]
(Converted to a discussion, as it is not known to be a bug yet).

How are you including Styles.xaml?

**Reply by ObsidianSilence** [reply]
In my App.xaml I added it as a merged dictionary, I tried to follow the sample project.
![image](https://user-images.githubusercontent.com/50368420/111194853-fdd0be00-8591-11eb-8ff2-90641accfd02.png)

**Reply by canton7** [reply]
Odd. Can you share a simple application which reproduces this issue?

Other things to try:

1. Define the style in the enclosing window / UserControl (no ResourceDictionaries)
2. Include the ResourceDictionary in the enclosing window / UserControl

**Reply by ObsidianSilence** [reply]
I can share the application, it's just a bunch of UserControls that open other colored UserControls in an attempt to figured out how this all works. I will try you suggestions.

**Reply by canton7** [reply]
The suggestions are more about narrowing down what the cause might be.

Sharing would help: that will let me debug it.

**Reply by ObsidianSilence** [reply]
[ProactiveConversion.zip](https://github.com/canton7/Stylet/files/6143397/ProactiveConversion.zip)

**Reply by canton7** [reply]
Cheers, I'll take a look soon, although it might be tomorrow I'm afraid.

**Reply by ObsidianSilence** [reply]
Thank you for taking a look, for now I'll try your other options and if all else fails go back to setting it on each button. I'm enjoying figuring out how everything works.

**Reply by canton7** [reply]
Which button specifically is causing the issue?

**Reply by ObsidianSilence** [reply]
I just went through the code and removed each individual reference to Style, the problem did not go away until I removed the last reference to style for the last <TextBlock> the last thing I removed was the <TextBlock> on ToView.xaml. I also removed the reference to Styles.xaml as a merged reference library. Oddly enough it works just fine with a reference to Colors.Xaml and Styles.Xaml so long as Styles.xaml is not referenced in any of the xaml.

Well. Now I am at a loss, after removing everything and building component by component, I then added the Style reference back and rebuilt it to verify what was going on. I have since added back every single reference that I removed and have not been able to replicate the issue. The application now builds and runs without any errors. Sorry for wasting your time.
[ProactiveConversion.zip](https://github.com/canton7/Stylet/files/6144107/ProactiveConversion.zip)
